client/components/AllProducts.js

```diff
@@ -18,81 +21,55 @@ export class AllProducts extends React.Component {
 
   render() {
     const loading = (
-      <div className="spinner-border text-secondary" role="status">
-        <span className="visually-hidden">Loading...</span>
+      <div className='spinner-border text-secondary' role='status'>
+        <span className='visually-hidden'>Loading...</span>
       </div>
     );
 
     const { products } = this.props;
 
-    const MapProducts = () => {
-      return products.map((product) => {
-        return (
-          <div className="col" key={product.id}>
-            <div className="card shadow-sm">
-              <img
-                className="card-img-top"
-                src={product.imageUrl}
-                alt="image of product"
-              />
-              <div className="card-body">
-                <h4 className="card-title">
-                  <Link to={`/products/${product.id}`}>
-                    {product.productName}
-                  </Link>
-                </h4>
-
-                <address className="card-subtitle">{product.price}</address>
-                <p className="card-text">{product.description}</p>
-                <div className="d-flex justify-content-between align-items-center bottom-buttons">
-                  <div className="btn-group">
-                    <Link
-                      type="button"
-                      className="btn btn-sm btn-outline-secondary"
-                      to={`/products/${product.id}`}
-                    >
-                      View
-                    </Link>
-                  </div>
-                  <button
-                    type="button"
-                    className="btn btn-danger"
-                    onClick={() => this.props.deleteProduct(product.id)}
-                  >
-                    X
-                  </button>
-                </div>
-              </div>
-            </div>
-          </div>
-        );
-      });
-    };
-
     return (
       <div className="all-products w-100">
-        <section className="py-5 text-center container">
-          <div className="row py-lg-5">
-            <div className="col-lg-6 col-md-8 mx-auto">
-              <h1 className="fw-light">
-                {this.state.loading && loading}All Products
-              </h1>
-              <p>
-                <Link to="/products/create" className="btn btn-primary my-2">
-                  Create a new product
-                </Link>
-              </p>
-            </div>
+        <div className="position-relative overflow-hidden p-3 p-md-5 m-md-3 text-center bg-light hero-image">
+          <div className="p-lg-5 mx-auto my-5">
+            <h1 className="display-4 fw-normal text-light">
+              {this.state.loading && loading}Shop
```

Comment:
I would just put the jsx directly in-line here, rather than defining it above. It still works like this, but I think it can be hard to understand that `loading` here is actually a variable that contains jsx